### PR TITLE
plat-stm32mp1: update STMP32MP13 DTS + etzpc/rstclk/mco-clock probing

### PR DESCRIPTION
**Pull request by etienne-lms**
Platform DTS and drivers updated needed to move the platform to PINCTRL framework.

### CODE REVIEW

File: core/arch/arm/dts/stm32mp135f-dk.dts [hunk 1]
```diff
@@ -87,6 +298,7 @@
 		CLK_STGEN_HSE
 		CLK_USBPHY_HSE
 		CLK_I2C4_HSI
+		CLK_I2C5_HSI
```
Comment (GseoC) [inline core/arch/arm/dts/stm32mp135f-dk.dts:301]: Won't this remain unused?

File: core/include/dt-bindings/clock/stm32mp13-clks.h [hunk 2]
```diff
@@ -193,7 +193,13 @@
 #define SAI1		160
 #define SAI2		161
 
-#define STM32MP1_LAST_CLK 162
+#define SPI1		162
```
Comment (GseoC) [inline core/include/dt-bindings/clock/stm32mp13-clks.h:196]: What is the purpose of adding these?

File: core/include/dt-bindings/rtc/rtc-stm32.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: GPL-2.0 or BSD-3-Clause */
+/*
+ * This header provides constants for STM32_RTC bindings.
```
Comment (GseoC) [inline core/include/dt-bindings/rtc/rtc-stm32.h:3]: Copyright's missing, maybe update header?

**Comment by etienne-lms** [comment]
comment addressed

File: core/include/dt-bindings/clock/stm32mp13-clks.h (hunk 2 above)
Comment (GseoC) [reply core/include/dt-bindings/clock/stm32mp13-clks.h:196]: I meant we do not have a use case but I guess it's completely fine for synchro

File: core/arch/arm/dts/stm32mp135f-dk.dts (hunk 1 above)
Comment (GseoC) [reply core/arch/arm/dts/stm32mp135f-dk.dts:301]: I meant we do not have a use case but I guess it's completely fine for synchro

File: core/drivers/clk/clk-stm32mp13.c [hunk 4]
```diff
@@ -2661,3 +2663,26 @@ static TEE_Result stm32mp13_clk_probe(const void *fdt, int node,
 }
 
 CLK_DT_DECLARE(stm32mp13_clk, "st,stm32mp13-rcc", stm32mp13_clk_probe);
+static TEE_Result stm32mp13_rcc_mco_probe(const void *fdt, int node,
```
Comment (patrickdelaunay) [inline core/drivers/clk/clk-stm32mp13.c:2666]: I prefer to managed MCO and critical clock in a sub node 
generic for each SOC of STM32MP family

=> no need to have a specific new compatible "st,stm32mp13-rcc-mco" only managed in OP-TEE for this feature

I will propose something soon for this part.

Patrick

File: core/arch/arm/dts/stm32mp135f-dk.dts (hunk 1 above)
Comment (patrickdelaunay) [reply core/arch/arm/dts/stm32mp135f-dk.dts:301]: st,stm32mp13-rcc-mco => not domcuement in yaml file, no reason to change compatible 

I prefer a other solution, can you drop the MCO part, I will propose something soon

Patrick

File: core/arch/arm/dts/stm32mp13-pinctrl.dtsi [hunk 5]
```diff
@@ -16,6 +16,15 @@
 		};
 	};
 
+	rcc_mco_pins_a: rcc-pins-0 {
```
Comment (patrickdelaunay) [inline core/arch/arm/dts/stm32mp13-pinctrl.dtsi:19]: see other reamrks => please drop MCO part

File: core/drivers/clk/clk-stm32mp13.c (hunk 4 above)
Comment (etienne-lms) [reply core/drivers/clk/clk-stm32mp13.c:2666]: Ok, I understand. I'll remove the MCO clock part from the series.

File: core/arch/arm/dts/stm32mp13-pinctrl.dtsi (hunk 5 above)
Comment (etienne-lms) [reply core/arch/arm/dts/stm32mp13-pinctrl.dtsi:19]: I'll leave the pinctrl part which is fine for that board. I'll only discard the MCO driver part and its node/compatible in the DT.

File: core/arch/arm/dts/stm32mp135f-dk.dts (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/dts/stm32mp135f-dk.dts:301]: @patrickdelaunay, I think your comment is misplaced and adding `CLK_I2C5_HSI` source clock is not related to MCO. Please tell me if i'm wrong.

**Comment by etienne-lms** [comment]
Comments addressed + added a fixup patch on I2C pin config.
@GseoC, @patrickdelaunay, please tell me if you want me to squash the fixup commits into a clean series.

**Comment by etienne-lms** [comment]
review tag applied
thanks

**Pull request merged**